### PR TITLE
Romeo should verify and update the public key on startup

### PR DESCRIPTION
**Pull request by stjepangolemac**
- [x] Will set the public key when contract is discovered
- [x] State handling refactored to make it clearer and more understandable
- [x] Logging cleaned up and now uses sane defaults (no need to set RUST_LOG to see something)

### CODE REVIEW

File: romeo/src/state.rs [hunk 1]
```diff
@@ -12,332 +12,637 @@ use sbtc_core::operations::{
...
-#[derive(Debug, Default, serde::Serialize, serde::Deserialize)]
-pub struct State {
-	deposits: Vec<Deposit>,
-	withdrawals: Vec<Withdrawal>,
-	stacks_block_height: Option<u32>,
-	bitcoin_block_height: Option<u32>,
-}
+/// Romeo internal state
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub enum State2 {
```
Comment (friedger) [inline romeo/src/state.rs:25]: Why `State2` and not `State` ?

**Review by friedger — DISMISSED** [review]
Looks good to me.

**Review by radicleart — APPROVED** [review]
Looks good to me.

**Comment by friedger** [comment]
Needs rebase :-)

File: romeo/src/state.rs (hunk 1 above)
Comment (stjepangolemac) [reply romeo/src/state.rs:25]: Leftover from the refactor, will fix.

**Comment by stjepangolemac** [comment]
> Looks good to me.
> 
> Q. Do we need to set the 33 byte public key? Given we always strip the first (x-only) byte to get the schnorr compatible one - is there a case where its needed?

You mean why not write 32 bytes only?

**Comment by stjepangolemac** [comment]
I've updated the branch and could use your reapprovals @radicleart and @friedger.

**Review by friedger — APPROVED** [review]
LGTM

**Review by radicleart — APPROVED** [review]
lgtm

**Comment by friedger** [comment]
@AshtonStephens Mike's review does not count.

**Comment by stjepangolemac** [comment]
> @AshtonStephens Mike's review does not count.

Does anyone know why is this happening?

**Comment by AshtonStephens** [comment]
> > @AshtonStephens Mike's review does not count.
> 
> Does anyone know why is this happening?

I've added Mike as a maintainer

**Pull request merged**